src/main/scala/coupledL2/tl2chi/MSHR.scala

```diff
@@ -262,13 +279,11 @@ class MSHR(implicit p: Parameters) extends TL2CHIL2Module with HasCHIOpcodes {
   val fwdCacheState = Mux(
     isSnpToBFwd(req_chiOpcode),
     SC,
-    Mux(
-      req_chiOpcode === SnpUniqueFwd,
-      Mux(meta.dirty || probeDirty, UD, UC),
-      I
-    )
+    Mux(isSnpToNFwd(req_chiOpcode), UC /*UC_UD*/, I)
   )
-  val fwdPassDirty = req_chiOpcode === SnpUniqueFwd && (meta.dirty || probeDirty)
+  val fwdPassDirty = isSnpToNFwd(req_chiOpcode) && (
+    (dirResult.hit && meta.dirty) || probeDirty ||
+    (req.snpHitRelease && req.snpHitReleaseWithData))
```

Comment:
Please use a new val to declare `(dirResult.hit && meta.dirty) || probeDirty || (req.snpHitRelease && req.snpHitReleaseWithData)` because it has appeared too many times. And remove unnecessary brackets please :)

Reply:
Refactored on https://github.com/OpenXiangShan/CoupledL2/pull/261/commits/bf620e1c00e916fc26db1fac2fb3ca988a47eba5 :D